File: go/LeetCode/maximumTime.go
package main

func main() {
	
}

func maximumTime(time string)  string {
	temp :=[]byte(time)
	if temp[0] == '?'{
		if temp[1]=='?' || temp[1]<'4' {
			temp[0]='2'
		}else {
			temp[0]='1'
		}
	}
	if temp[1] == '?' {
		if temp[0] == '2' {
			temp[1] = '3'
		}else {
			temp[1] = '9'
		}
	}
	if temp[3]=='?' {
		temp[3]='5'
	}
	if temp[4]=='?' {
		temp[4]='9'
	}
	return string(temp)
}

File: python3/LeetCode/maximumTime.py
class Solution:
    def maximumTime(self, time:str) ->str:
        [a, b, _, c, d] = time
        if a =='?':
            a=='2' if b=='?' or b <'4' else '1'
        if b=='?':
            b='3' if a=='2' else '9'
        if c=='?':
            c='5'
        if d=='?':
            d='9'
        return f'{a}{b}:{c}{d}'


if __name__ == '__main__':
    cache = dict()
    i=1
    m=1
    cache[(i, m)] =1
    print(cache)